java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -1051,12 +1057,17 @@ private void alignPackageCache(Channel channel, Set<Package> oldChannelPackages)
         // add cache entries for new ones
         Set<Package> newTgtPackages = new HashSet<>(channel.getPackages());
         newTgtPackages.removeAll(oldChannelPackages);
-        ErrataCacheManager.insertCacheForChannelPackages(channel.getId(), null, extractPackageIds(newTgtPackages));
+        ErrataCacheManager.insertCacheForChannelPackages(
+                channel.getId(),
+                null,
+                extractPackageIds(newTgtPackages)
+        );
     }
 
     // helper for extracting certain filter types
     // it's not optimal to run this method multiple times for same collection of filters, but at least it's clear
-    private static <T> List<T> extractFiltersOfType(Collection<ContentFilter> filters, Class<T> type) {
+    private static <T> List<T> extractFiltersOfType(Collection<ContentFilter<? extends ContentFilterEntity>> filters,
+                                                    Class<T> type) {
         return filters.stream()
                 .filter(f -> type.isAssignableFrom(f.getClass()))
                 .map(f -> (T) f)
```

Comment:
I know this was not part of your change, but to avoid the unchecked warning this should be:

```java
                .map(f -> type.cast(f))
```